src/MDMScheduler.m

```diff
@@ -81,16 +81,33 @@ - (MDMSchedulerActivityState)activityState {
 }
 
 - (void)addPlan:(NSObject<MDMPlan> *)plan to:(id)target {
+  [self commonAddPlan:plan toTarget:target log:nil];
+}
+
+- (void)addPlan:(NSObject<MDMPlan> *)plan named:(NSString *)name to:(id)target {
+  NSParameterAssert(name.length > 0);
+  [self commonAddPlan:plan toTarget:target log:[[MDMTransactionLog alloc] initWithPlans:@[plan] target:target name:name removal:FALSE]];
```

Comment:
Hmmmmm - I think there's a `PlanInfo` class in the Android codebase which acts as a data holder. I was wondering whether we could just move all this metadata into the `MDMPlan` itself instead of keeping track of a plan and metadata associated with a plan in two separate objects?